keel-tagging-plugin/src/main/kotlin/com/netflix/spinnaker/keel/tagging/KeelTagSpec.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ *
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.netflix.spinnaker.keel.tagging
+
+import com.netflix.spinnaker.keel.tags.EntityRef
+import com.netflix.spinnaker.keel.tags.EntityTag
+
+/**
+ * The desired for Spinnaker tagging of resources we manage
+ *
+ * If [keelTagged] is true the resource should only have tag
+ * in the [KEEL_TAG_NAMESPACE], and that tag should match the given tag.
+ *
+ * If [keelTagged] is false the resource should have no tags in
+ * the [KEEL_TAG_NAMESPACE].
+ */
+data class KeelTagSpec(
+  val keelId: String,
+  val entityRef: EntityRef,
+  val tagState: TagState
+)
+
+data class TaggedResource(
+  val keelId: String,
+  val entityRef: EntityRef,
+  val relevantTag: EntityTag?
+)
+
+sealed class TagState
```

Comment:
Am I right in thinking we desire a tag on a managed resource, and remove it from anything that gets deleted?

Reply:
yes